source/2015-11-27-colorize-svgs-with-sass.html.markdown

```diff
@@ -0,0 +1,80 @@
+---
+title: Colorize SVGs with SASS
+date: 2015-11-27 16:05 UTC
+author: bro
+tags: 
+---
+
+As you might know, it is possible to style an SVG graphic via CSS if you
+inline it. The problem, through, is that you cannot do this when the SVG
+image is used as a background-image in CSS.
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<svg xmlns="http://www.w3.org/2000/svg" width="400" height="400">
+  <circle cx="100" cy="100" r="50" stroke="black" stroke-width="5" fill="none" />
+</svg>
+```
+
+```css
+.circle {
+  background-image url(circle.svg);
+}
+```
+
+You can use [CSS masks](http://www.w3.org/TR/css-masking/), but not all
+browsers [support it](http://caniuse.com/#feat=css-masks).
+
+```css
+.circle {
+  background-color: blue;
+  mask-image: url(circle.svg);
+}
+```
+
+But if you use SASS, you can write your own SASS function to inline the SVG
+and for example change the color of all strokes (script inspired by 
+[this](https://github.com/Compass/compass/issues/1460#issuecomment-65132746)
+Compass issue):
+
+```ruby
+require 'sass'
+require 'cgi'
+
+module Sass::Script::Functions
+  def colorize_svg(svg_name, color)
+    raise "Not a valid color: #{color}" unless color.is_a?(Sass::Script::Color)
+
+    asset = Rails.application.assets.find_asset(svg_name.value)
+    raise "File not found: #{svg_name}" unless asset
+
+    svg = File.open(asset.pathname, "rb") { |io| io.read }
```

Comment:
`svg = File.read(asset.pathname`, but I guess we're not really doing code reviews here ;-)